src/headless/converse-muc.js

```diff
@@ -941,74 +942,38 @@ converse.plugins.add('converse-muc', {
                 ])].filter(n => n);
             },
 
-            getReferenceForMention (mention, index) {
-                const nicknames = this.getAllKnownNicknames();
-                const longest_match = u.getLongestSubstring(mention, nicknames);
-                if (!longest_match) {
-                    return null;
-                }
-                if ((mention[longest_match.length] || '').match(/[A-Za-zäëïöüâêîôûáéíóúàèìòùÄËÏÖÜÂÊÎÔÛÁÉÍÓÚÀÈÌÒÙ0-9]/i)) {
-                    // avoid false positives, i.e. mentions that have
-                    // further alphabetical characters than our longest
-                    // match.
-                    return null;
-                }
+            getOccupantByNickname (nickname) {
+                return this.occupants.findOccupant({ nick: nickname })
+                    || u.isValidJID(nickname)
+                    && this.occupants.findOccupant({ jid: nickname });
+            },
 
-                let uri;
-                const occupant = this.occupants.findOccupant({'nick': longest_match}) ||
-                        u.isValidJID(longest_match) && this.occupants.findOccupant({'jid': longest_match});
+            parseTextForReferences (originalMessage) {
```

Comment:
Please stick to the convention in this codebase of writing variable names all lower-case with underscores.
See here: https://conversejs.org/docs/html/style_guide.html#underscores-or-camelcase